### PR TITLE
Add performance tests

### PR DESCRIPTION
**Pull request by pivnicek**
related to #31

### CODE REVIEW

File: tests/Keboola/DbExtractor/AbstractMSSQLTest.php [hunk 1]
```diff
@@ -120,6 +120,13 @@ protected function createTextTable(CsvFile $file, ?array $primaryKey = null, ?st
             $tableName = $overrideTableName;
         }
 
+        try {
+            $this->pdo->exec(sprintf("DROP TABLE %s", $tableName));
```
[the remark this answers is not in the document]
Comment (tomasfejfar) [reply tests/Keboola/DbExtractor/AbstractMSSQLTest.php]: also extract to method

File: tests/Keboola/DbExtractor/MSSQLTest.php [hunk 2]
```diff
@@ -1260,31 +1260,46 @@ public function testStripNulls(): void
...
+                echo "\ncreating table $tableCount\n";
+                $this->pdo->exec(
+                    sprintf(
+                        "CREATE TABLE [testschema_%d].[testtable_%d] ([ID] INT IDENTITY(1,1) NOT NULL%s)",
+                        $schemaCount,
+                        $tableCount,
+                        $columnsSql
+                    )
+                );
+                $this->pdo->exec(
```
Comment (tomasfejfar) [inline tests/Keboola/DbExtractor/MSSQLTest.php]: is it more efficient then just adding a PRIMARY KEY to the ID field?

File: tests/Keboola/DbExtractor/MSSQLTest.php [hunk 3]
```diff
@@ -1260,31 +1260,46 @@ public function testStripNulls(): void
...
-        sleep(5);
 
-        $this->pdo->exec($createTableScript);
-        // wait for the create script to finish
-        sleep(5);
+        for ($schemaCount = 0; $schemaCount < $numberOfSchemas; $schemaCount++) {
+            echo "\ncreating schema $schemaCount\n";
+            $this->pdo->exec(sprintf("CREATE SCHEMA [testschema_%d]", $schemaCount));
+            for ($tableCount = 0; $tableCount < $numberOfTablesPerSchema; $tableCount ++){
+                echo "\ncreating table $tableCount\n";
```
Comment (tomasfejfar) [inline tests/Keboola/DbExtractor/MSSQLTest.php]: sounds very verbose :) With 1000 tables I'd opt in for dots instead to show progress...

File: tests/Keboola/DbExtractor/AbstractMSSQLTest.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (tomasfejfar) [reply tests/Keboola/DbExtractor/AbstractMSSQLTest.php]: ideally with optional schema and use it when cleaning up generated tables

File: tests/Keboola/DbExtractor/AbstractMSSQLTest.php [hunk 4]
```diff
@@ -120,6 +120,13 @@ protected function createTextTable(CsvFile $file, ?array $primaryKey = null, ?st
...
+        } catch (\Throwable $e) {
+            // table didn't exist
```
[the remark this answers is not in the document]
Comment (tomasfejfar) [reply tests/Keboola/DbExtractor/AbstractMSSQLTest.php]: Not very reliable IMHO.

File: tests/Keboola/DbExtractor/AbstractMSSQLTest.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (pivnicek) [reply tests/Keboola/DbExtractor/AbstractMSSQLTest.php]: ah, yeah, I'm doing it properly elsewhere.  dik!

File: tests/Keboola/DbExtractor/AbstractMSSQLTest.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (pivnicek) [reply tests/Keboola/DbExtractor/AbstractMSSQLTest.php]: yes, I'll replace with the correct way

File: tests/Keboola/DbExtractor/MSSQLTest.php (hunk 2 above)
Comment (pivnicek) [reply tests/Keboola/DbExtractor/MSSQLTest.php]: probably not, I'll merge the 2 statements.

**Comment by pivnicek** [comment]
Thanks @tomasfejfar !  I agree with your comment, I'll let you know when I've got it updated.

**Comment by pivnicek** [comment]
btw https://github.com/keboola/db-extractor-common/issues/32

**Comment by pivnicek** [comment]
@tomasfejfar I separated the performance tests into their own testsuite for now, this way we can easily make them run only on tags later if we want but for now imho it's not necessary because there is only the one running and it only takes just over a minute.  Otherwise, I believe I covered the rest.

**Comment by pivnicek** [comment]
Sorry @tomasfejfar , the large table test is a work in progress, this PR is mainly about the `getTables` performance. 

I've changed it to have a more appropriate message.

File: tests/Keboola/DbExtractor/MSSQLTest.php [hunk 5]
```diff
@@ -1225,146 +1225,6 @@ public function testStripNulls(): void
...
-        $app = $this->createApplication($config);
-        $startTime = time();
-        $result = $app->run();
-        $this->assertEquals('success', $result['status']);
-        $runTime = time() - $startTime;
-
-        echo "\nThe app ran in " . $runTime . " seconds.\n";
-
-        $this->pdo->exec("IF OBJECT_ID('dbo.largetest', 'U') IS NOT NULL DROP TABLE dbo.largetest");
+        $this->dropTable("NULL_TEST");
```
[the remark this answers is not in the document]
Comment (pivnicek) [reply tests/Keboola/DbExtractor/MSSQLTest.php]: Looking at the full file, the names seem correct to me for each method.

**Comment by tomasfejfar** [comment]
So why push it in this PR, you can save it for later if it's WIP...

File: tests/Keboola/DbExtractor/MSSQLTest.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (tomasfejfar) [reply tests/Keboola/DbExtractor/MSSQLTest.php]: oh, I mislooked - it was incorrect before :D Now it's fixed.

File: tests/Keboola/DbExtractor/MSSQLTest.php [hunk 6]
```diff
@@ -838,7 +837,6 @@ public function testGetTables(): void
                                     'type' => 'varchar',
                                     'length' => 55,
                                     'nullable' => false,
-                                    'default' => '(\'mario\')',
```
Comment (tomasfejfar) [inline tests/Keboola/DbExtractor/MSSQLTest.php]: this is suspicious - it had default before (which was loaded straight from array) now it disappeared. Why?

File: tests/Keboola/DbExtractor/MSSQLTest.php [hunk 7]
```diff
@@ -815,7 +816,6 @@ public function testGetTables(): void
                                     'type' => 'datetime',
                                     'length' => null,
                                     'nullable' => true,
-                                    'default' => '(\'2018-08-14 10:43:18\')',
```
Comment (tomasfejfar) [inline tests/Keboola/DbExtractor/MSSQLTest.php]: how is it that default disappeared?

Comment (pivnicek) [reply tests/Keboola/DbExtractor/MSSQLTest.php]: ah, the getTables does not return defaults in the new version

Comment (pivnicek) [reply tests/Keboola/DbExtractor/MSSQLTest.php]: it's way too complicated via the `sys` tables and unnecessary information for the consumer

Comment (tomasfejfar) [reply tests/Keboola/DbExtractor/MSSQLTest.php]: if you say so - just that it changed - that's all. I don't know where the data is consumed...

**Pull request merged**